quelpa.el

```diff
@@ -371,9 +422,14 @@ If there is an error and no existing checkout return nil."
 (defun quelpa-get-melpa-recipe (name)
   "Read recipe with NAME for melpa git checkout.
 Return the recipe if it exists, otherwise nil."
-  (let* ((recipes-path (expand-file-name "package-build/recipes" quelpa-build-dir))
+  (let* ((recipes-path (car quelpa-melpa-recipe-dirs))
          (files (directory-files recipes-path nil "^[^\.]+"))
          (file (assoc-string name files)))
```

Comment:
with `cl-loop` it's quite clean:

```cl
(defun quelpa-get-melpa-recipe (name)
  "Read recipe with NAME for melpa git checkout.
Return the recipe if it exists, otherwise nil."
  (cl-loop for dir in quelpa-melpa-recipe-dirs
           for file = (assoc-string name (directory-files dir nil "^[^\.]+"))
           when file
           return (with-temp-buffer
                    (insert-file-contents-literally
                     (expand-file-name file dir))
                    (read (buffer-string)))))
```

this version also respects the order of the directories. it returns the first recipe found